admin/customers.php

```diff
@@ -168,17 +159,17 @@
 
             $zone_id = 0;
             $entry_state_error = false;
-            if (ACCOUNT_STATE == 'true') {
+            if (ACCOUNT_STATE === 'true') {
                 $entry_state_has_zones = count(zen_get_country_zones($entry_country_id)) > 0;
                 if ($entry_state_has_zones) {
                     $zone_query = $db->Execute(
                         "SELECT zone_id
                            FROM " . TABLE_ZONES . "
-                           WHERE zone_country_id = " . (int)$entry_country_id . "
+                           WHERE zone_country_id = " . $entry_country_id . "
```

Comment:
I've always preferred to keep the cast, even if redundant, inside SQL queries ... because then there's never any doubt about whether it's been properly sanitized .... and it doesn't require searching the entire file or the files that called this file in order to find out whether it was already the right type, and that 3rd party plugin code can not have tampered with it in any way that could leave any bSLQi vulnerabilities.

Reply:
I agree on string-related values which need to be properly sanitized, but the processing prior to this (and the following) SQL statements has guaranteed that the values are already of an integer type.  If some rogue add-on has made modifications to set one or more of the values to a non-integer value, the SQL query/update will fail.

Reply:
>  If some rogue add-on has made modifications to set one or more of the values to a non-integer value, the SQL query/update will fail.

It may fail, but if the non-integer value happens to be a string containing a rogue SQL injection, while the intended core code payload may fail, unintended side-effects may take place even without knowing.